rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,424 @@
+            )
+        ):
+            log.warning(
+                f'Queried Alchemy API from {from_asset.identifier} to {to_asset.identifier}, '
+                f'but received mismatched symbol in response',
+            )
+            return ZERO_PRICE
+
+        if result['currency'] != 'usd':
+            log.warning(
+                f'Queried Alchemy API from {from_asset.identifier} to {to_asset.identifier}, '
+                f'but received non-USD currency in response',
+            )
+            return ZERO_PRICE
+
+        if not (data := result.get('data', [])):
+            log.warning(
+                f'Queried Alchemy API from {from_asset.identifier} to {to_asset.identifier}, '
+                f'but received no historical price data in response',
+            )
+            return ZERO_PRICE
+
+        try:
+            return deserialize_price(data[-1]['value'])
+        except (DeserializationError, KeyError, IndexError) as e:
+            error_msg = f'missing key {e!s}' if isinstance(e, KeyError) else str(e)
+            log.warning(
+                f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}. '  # noqa: E501
+                f'Failed to deserialize historical price data due to: {error_msg}',
+            )
+            return ZERO_PRICE
+
+    @overload
+    def _query(
+            self,
+            verb: Literal['GET'],
+            payload: dict[str, Any],
+            endpoint: Literal['by-symbol'],
+    ) -> dict[str, Any]: ...
+
+    @overload
+    def _query(
+            self,
+            verb: Literal['POST'],
+            payload: dict[str, Any],
+            endpoint: Literal['by-address'],
+    ) -> dict[str, Any]: ...
+
+    @overload
+    def _query(
+            self,
+            verb: Literal['POST'],
+            payload: dict[str, Any],
+            endpoint: Literal['historical'],
+    ) -> dict[str, Any]: ...
+
+    def _query(
+            self,
+            verb: Literal['GET', 'POST'],
+            payload: dict[str, Any],
+            endpoint: Literal['by-symbol', 'by-address', 'historical'],
+    ) -> dict[str, Any]:
+        if (api_key := self._get_api_key()) is not None:
+            base_url = f'https://api.g.alchemy.com/prices/v1/{api_key}/tokens'
+        else:
+            base_url = 'https://api.g.alchemy.com/prices/v1/Y_nFGh5UCi30NzRWOAam0gsVpQrjDXZX/tokens'
+
+        url = f'{base_url}/{endpoint}'
+        log.debug(f'Querying alchemy api: {url=} with {payload=}')
+        try:
+            if verb == 'GET':
+                response = self.session.get(
+                    url=url,
+                    params=payload,
+                    timeout=CachedSettings().get_timeout_tuple(),
+                )
+            else:  # POST
+                response = self.session.post(
+                    url=url,
+                    json=payload,
+                    timeout=CachedSettings().get_timeout_tuple(),
+                )
```

Comment:
This can be simplified I believe

``` 
self.session.request(
    method=verb,
    url=url,
    **{'params' if verb == 'GET' else 'json': payload},
    timeout=CachedSettings().get_timeout_tuple(),
)
```